pkg/k8sutil/k8sutil.go

```diff
@@ -115,3 +117,42 @@ func GetPod(ctx context.Context, client crclient.Client, ns string) (*corev1.Pod
 
 	return pod, nil
 }
+
+// GetGVKsFromAddToScheme takes in the runtime scheme and filters out all generic apimachinery meta types.
+// It returns just the GVK specific to this scheme.
+func GetGVKsFromAddToScheme(addToSchemeFunc func(*runtime.Scheme) error) ([]schema.GroupVersionKind, error) {
+	s := runtime.NewScheme()
+	err := addToSchemeFunc(s)
+	if err != nil {
+		return nil, err
+	}
+	schemeAllKnownTypes := s.AllKnownTypes()
+	ownGVKs := []schema.GroupVersionKind{}
+	for gvk, _ := range schemeAllKnownTypes {
+		if !isKubeMetaKind(gvk.Kind) {
+			ownGVKs = append(ownGVKs, gvk)
+		}
+	}
+
+	return ownGVKs, nil
+}
+
+func isKubeMetaKind(kind string) bool {
+	if strings.HasSuffix(kind, "List") ||
+		kind == "GetOptions" ||
```

Comment:
It would be easier to manage these kind strings in a package-level `var metaTypes []string`, and iteratively check for equality.

Reply:
See above discussion, moved per @joelanford suggestion :)